library/std/src/sys/unix/mod.rs

```diff
@@ -86,6 +96,28 @@ pub fn init() {
         reset_sigpipe();
     }
 
+    #[cfg(not(any(target_os = "emscripten", target_os = "fuchsia", target_os = "redox")))]
+    unsafe fn sanitize_standard_fds() {
+        let pfds: &mut [_] = &mut [
+            libc::pollfd { fd: 0, events: 0, revents: 0 },
+            libc::pollfd { fd: 1, events: 0, revents: 0 },
+            libc::pollfd { fd: 2, events: 0, revents: 0 },
+        ];
+        if libc::poll(pfds.as_mut_ptr(), 3, 0) == -1 {
+            return;
+        }
+        for pfd in pfds {
+            if pfd.revents & libc::POLLNVAL == 0 {
+                continue;
+            }
+            if libc::open("/dev/null\0".as_ptr().cast(), libc::O_RDWR, 0) == -1 {
```

Comment:
```suggestion
            let mut pipe_fds = [0; 2];
            if libc::pipe(pipe_fds.as_mut_ptr()) == -1 {
```

And then before the end of the loop:
```rust
if pfd == 0 {
    // The read end of the pipe is already fd 0
    libc::close(pipe_fds[1]);
} else {
    // dup2 the write end to the desired fd (1 or 2), which will close the read end
    libc::dup2(pipe_fds[1], pfd);
    // close the old write end
    libc::close(pipe_fds[1]);
}
```
plus error handling.

Reply:
When standard file descriptors are closed the stdin / stdout / stderr are
required not to panic per RFC 1014. The closed pipe variant is not an option as
proposed.

Reply:
Interesting. I wasn't aware of RFC 1014. Given that, I agree, this won't work. Objection withdrawn; let's go with /dev/null.